arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceViewController.swift

```diff
@@ -0,0 +1,404 @@
+        case .conditions:
+            return traceConditionalExpressions.count
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch Section.allCases[indexPath.section] {
+        case .switches:
+            let cell = tableView.dequeueReusableCell(withIdentifier: "SwitchCell", for: indexPath)
+            switch Switches.allCases[indexPath.row] {
+            case .barriers:
+                cell.accessoryView = barriersSwitch
+                cell.textLabel?.text = Switches.barriers.label
+            case .containers:
+                cell.accessoryView = containersSwitch
+                cell.textLabel?.text = Switches.containers.label
+            }
+            return cell
+        case .conditions:
+            let cell = tableView.dequeueReusableCell(withIdentifier: "ConditionCell", for: indexPath)
+            cell.textLabel?.text = expressionToString(expression: traceConditionalExpressions[indexPath.row])
+            return cell
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+        if indexPath.section == Section.conditions.rawValue && indexPath.row != 0 {
+            // Enable editing for the rest of conditions section.
+            return true
+        } else {
+            // Disable editing for the first row of conditions section and other rows.
+            return false
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if indexPath.section == Section.conditions.rawValue && editingStyle == .delete {
+            traceConditionalExpressions.remove(at: indexPath.row)
+            tableView.deleteRows(at: [indexPath], with: .fade)
+            // Reload footer label.
+            chainedConditionsLabel.text = expressionString
+        }
+    }
+}
+
+/// An extension of `AGSUtilityCategoryComparisonOperator` that returns a human readable description.
+/// - Note: You may also create a `AGSUtilityCategoryComparison` with
+///         `AGSUtilityNetworkDefinition.categories` and `AGSUtilityCategoryComparisonOperator`.
+private extension AGSUtilityCategoryComparisonOperator {
+    var title: String {
+        switch self {
+        case .exists: return "Exists"
+        case .doesNotExist: return "DoesNotExist"
+        @unknown default: return "Unknown"
+        }
+    }
+}
+
+/// An extension of `AGSUtilityAttributeComparisonOperator` that returns a human readable description.
+extension AGSUtilityAttributeComparisonOperator: CaseIterable {
+    // Manually set the maximum raw value of cases to facilitate protocol conformation.
+    private static let maxCaseValue = Self.doesNotIncludeAny.rawValue
+    
+    public static let allCases = {
+        return (0...maxCaseValue).map { AGSUtilityAttributeComparisonOperator(rawValue: $0)! }
```

Comment:
This is a clever idea, but there's more resilient ways of getting all the cases, assuming the raw values begin at `0` are are sequential. Here's one option:

```swift
    static let allCases = Self(rawValue: 0).map { (firstCase) in
        Array(sequence(first: firstCase, next: { Self(rawValue: $0.rawValue + 1) }))
    } ?? []
```

Reply:
Thanks! I actually tried a similar approach (less fancy, of course 😏), but found out that NS Enum does not give nil when init called with an invalid raw value. See the screenshot for rawValue >= 0x9

![image](https://user-images.githubusercontent.com/9660181/92286464-faaa9600-eebb-11ea-9077-ad49627d9581.png)

Therefore, the array will keep growing until crashing the app, when you tap on the comparison row in the options.

Further research told me NS Enum are similar to C enum that deal with unsafe pointers. I don't feel confident about unsafe, so I just used this uglier solution instead.

Reply:
That is fascinating. I thought for sure that used to work, but maybe I'm remembering incorrectly.